### PR TITLE
Switch to klee/klee and update expected test outputs

### PR DESCRIPTION
**Pull request by jamesreaver**

### CODE REVIEW

**Comment by jamesreaver** [comment]
Do you have any comments on these commits or can they be merged? @andronat

File: src/klee_web/worker/tests/test_runner.py [hunk 1]
```diff
@@ -60,13 +60,17 @@ def test_symargs(self):
             }
         })
 
+    # This test must now be run with the sym-stdin flag.
+    # When support for this flag is added, this test
+    # can be uncommented and updated.
+    """
```
Comment (andronat) [inline src/klee_web/worker/tests/test_runner.py]: Don't use `"""` to comment code. `"""` are used only for documentation purposes. Use `#`.

File: src/klee_web/worker/tests/test_runner.py [hunk 2]
```diff
@@ -60,13 +60,17 @@ def test_symargs(self):
...
     def test_symfiles(self):
```
Comment (andronat) [inline src/klee_web/worker/tests/test_runner.py]: If I understand correctly you commented out this test because the `run_klee_test` method doesn't support `--sym-stdin` in klee? If that's the case then the code of  `run_klee_test` should refactored to make the test pass.

**Comment by andronat** [comment]
We will freeze all merging-s until we make CircleCI pass

**Comment by andronat** [comment]
This needs a rebase.

File: src/klee_web/worker/tests/test_runner.py (hunk 2 above)
Comment (jamesreaver) [inline src/klee_web/worker/tests/test_runner.py]: This test tests symbolic standard input, but it does that using symbolic files (with file descriptor 0). This has changed with recent versions of klee, so that the flag `--sym-stdin` is used instead to make the standard input symbolic. For this reason, this test needs to be updated to use the flag, but we need another branch to be merged before using the flag. What shall we do with this test until then?

Comment (andronat) [inline src/klee_web/worker/tests/test_runner.py]: Which PR are we depending on?

Comment (jamesreaver) [inline src/klee_web/worker/tests/test_runner.py]: There isn't a PR for the branch that adds `--sym-stdin`. My plan is to rebase that branch, make sure  `--sym-stdin`  works correctly, and update this test as well as adding more. But at the moment that test is checking something which belong to an old version of klee, so rather than removing it I commented it out. I will come back at it in the other branch. Is that okay?

File: src/klee_web/worker/tests/test_runner.py (hunk 1 above)
Comment (jamesreaver) [inline src/klee_web/worker/tests/test_runner.py]: Thank you for the tip! I've fixed it now :)

**Comment by jamesreaver** [comment]
@andronat The last commit (and probably the ones coming) are only to test and debug CircleCI. When I find out why it was failing, I will rebase it all nicely. Unless I'm wrong, it's the only way, since these are issues that only appear in CircleCI.

**Comment by andronat** [comment]
@giacomoguerci I understand what you say but the problem with this is that when CircleCI runs, it also includes functionality fixed with your current PR. The problem we have atm is that CircleCI does/doesn't work on the master branch. So if you want to test if your solution fixes the master branch you should isolate to only the master branch and not on top of a WIP (work in progress). Do I make sense? :/

**Comment by andronat** [comment]
@giacomoguerci Also let me include an example. Think of a case that your current PR also include a non deterministic bug. When you want to fix the problem on the master branch and you include your fix on the current PR then you might see CircleCI fail again for a different reason not related to your fix. This is why it is very important to keep code/fixes etc separated.

**Comment by jamesreaver** [comment]
Oh I see what you mean. This PR is doing two things, switching to klee/klee, but also trying to fix the non-determinism in the tests of the master branch. I will remove the last commit from this PR, solve the problem with Nightmare's wait() on another branch, then, after that fix will be merged with master, we will continue with this PR which only switches to klee/klee. Do you agree with this plan, @andronat?

**Comment by andronat** [comment]
Yea that sounds good!

File: src/klee_web/worker/tests/test_runner.py (hunk 2 above)
Comment (andronat) [inline src/klee_web/worker/tests/test_runner.py]: Sorry for my late response. This PR #78 updates klee to the latest version right? This latest version supports `--sym-stdin` right? So if that's the case I think that you should fix this test case in this PR together with adding more tests.

Comment (jamesreaver) [inline src/klee_web/worker/tests/test_runner.py]: The latest version of Klee does support `--sym-stdin`, but the KleeWeb docker image doesn't yet (it's a different PR which I will open as soon as this one is merged). This is because the tests are not run by passing the flag directly, but by passing a configuration. The way KleeWeb reads this configuration will be updated in the other PR (at the moment it wouldn't know when to use the newly intorduced `--sym-stdin` flag).

Comment (andronat) [inline src/klee_web/worker/tests/test_runner.py]: But we don't want to keep KleeWeb. We want to move completely to `klee/klee`. I'm confused.

Comment (jamesreaver) [inline src/klee_web/worker/tests/test_runner.py]: Sorry, by by KleeWeb I meant klee web, this project. Not, kleeweb/klee, nothing to do with that.

Comment (andronat) [inline src/klee_web/worker/tests/test_runner.py]: Ok, so then I suggest to add the work you are doing on fixing how the `worker` parses the test through `run_klee_test` in this PR. Every PR should be a complete work that take care of all the problems introduced by a change. It's not a good idea to comment out a test that is not working because we made a change and then fix the test in another PR. Image a case where someone does the same and then never returns to fix the test case for various reasons.

Comment (jamesreaver) [inline src/klee_web/worker/tests/test_runner.py]: I think it's best to remove the test in this PR, given that it's testing a feature that is not part of this PR. This way this PR would be a complete work. I suggest keeping the changes needed to support `--sym-stdin` in a separate PR, because it's quite a big change and affects several files.

Comment (andronat) [inline src/klee_web/worker/tests/test_runner.py]: Ok, just put then a `TODO: Needs to be fixed` on top of it.

Comment (jamesreaver) [inline src/klee_web/worker/tests/test_runner.py]: All done! Tomorrow I will open a PR with support for `--sym-stdin` and relative tests.
